### PR TITLE
improve disk usage of zap

### PR DESCRIPTION
**Pull request by mschoch**
Breaking change, to be released as new major version (v13.0.0)

One of the optimizations in v12 was to encode a special marker
value when a field/term had no location information stored.
Unfortunately that marker was math.MaxUint64, which when
encoded via Uvarint requires 10 bytes.  In practice, the number
of bytes required went up, not down as a result of this change.

Now, we use 0 as that special marker value.  This has been
determined to be safe as 0 could never be the actual address
of term location information, as there are other mandatory
sections earlier in the file (stored field index).

### CODE REVIEW

**Comment by mschoch** [comment]
Still needs more testing, and some bleve release planning discussion.

**Comment by mschoch** [comment]
In the same file that was originally used to identify the problem (very small) the overall sizes were:

```
v11: 406
v12: 426
v13: 399
```

This looks minor, but it can really add up if you have lots of terms without term vector information.

Using `bleve-blast` with 100k docs, and 1000 docs/batch (no forced merging, so there is some noise depending on where merging was at the time indexing completed):

```
v11: 798M
v12: 1.6G
v13: 631M
```

**Review by sreekanth-cb — CHANGES_REQUESTED** [review]
Perhaps we need to consider this change for docValues too?
ref- https://github.com/blevesearch/zap/blob/fa29047c37898c01e253b3265ff6031fed8cd280/build.go#L29

**Review by sreekanth-cb — COMMENTED** [review]
As this is at a field level marker for docValues, savings won't be significant. But it adds some consistency in the way we represent "non encoded" values?

**Comment by fproulx-dfuse** [comment]
Is there any chance that this will get merged very soon to address the builder bug ?
We had set aside time to reprocess our indices with the new version, but now this is a showstopper.

Thanks.

**Comment by mschoch** [comment]
@fproulx-dfuse I had planned to do some more testing before merging it, but given that Bleve 1.0.x does not default to scorch, and if you choose scorch it still defaults to v11, users have to go out of there way to enable this.  So I am more comfortable moving forward wit that in mind.

That said, I would still like to get approval from @sreekanth-cb first so we maintain our typical review process.

**Review by sreekanth-cb — APPROVED** [review]

**Comment by mschoch** [comment]
@fproulx-dfuse sorry there will be a delay getting this merged, I have run into an issue in my testing this morning.

**Review by sreekanth-cb — APPROVED** [review]

**Review by sreekanth-cb — APPROVED** [review]

**Comment by mschoch** [comment]
We found additional problems related to the docValues change.  As that did not contribute significantly to the disk space usage, we have reverted that portion of the change, so that we can get the more important fix merged promptly.

**Review by sreekanth-cb — APPROVED** [review]

**Pull request merged**
